src/coreclr/jit/lowerxarch.cpp

```diff
@@ -130,6 +130,21 @@ void Lowering::LowerStoreIndir(GenTreeIndir* node)
             return;
         }
     }
+    else if (!node->HasIndex() && node->gtGetOp2()->OperIs(GT_CNS_DBL) &&
+             (FloatingPointUtils::convertDoubleToUInt64(node->gtGetOp2()->AsDblCon()->gtDconVal) == 0))
```

Comment:
Does `FloatingPointUtils::convertDoubleToUInt64` do what we want? Last time I looked at it, it was rounding cast...
There is a helper `IsCnsNonZeroFltOrDbl`, although it is awkwardly inverted (maybe worth adding a new one).